### ISSUE TITLE
CI updates

### ISSUE THREAD

**Issue by rhornung67** [LLNL/RAJA#1335]
Tasks to complete after 10/2022 release:

- [x] Switch Gitlab CI to 'rajasa' service user account
        - We need to get the group membership of the service user account fixed first.
- [x]  Update Dev Guide CI discussion to be consistent with recent switch to shared RADIUSS CI stuff
- [x] **More changes coming!!** Coordinate with Umpire (and other relevant projects) and update compiler specs in shared Gitlab CI configurations -- specifically, reduce shared pipelines to minimum needed
     - [x] Modify RADIUSS Spack Configs as needed
     - [x] Modify RADIUSS Shared CI as needed
     - [x] Update RADIUSS Spack Configs submodule in RAJA
- [x] Add OpenMP target offload CI checks on lassen (clang and xl)
- [ ] Add clang-cuda configuration on lassen
- [ ] Add checks to build with desul atomics submodule and external desul install (lassen, ruby, corona/tioga)
       - [ ] Note updating desul requires an update to BLT -- waiting on a new release.
- [ ] Add check for separable compilation for GPU back-ends (to prevent future breakages)
- [x] Add tioga CI checks for HIP 
- [x] Add tioga CI checks for HIP + OpenMP
- [x] Add local SYCL CI checks on corona (update compiler build to newer Intel and install in /usr/gapps to use in Gitlab CI)
- [ ] Add test piplelines to build with SIMD/SIMT support both on and off on each platform
- [x] Add newer gcc compiler versions, like 11.x, on azure that aren't available on LC systems so we can catch issues with SIMD vectorization machinery, for example. Coordinate with rse-ops container repo.
- [x] Add Gitlab CI for poodle (CZ CTS-2 machine) if/when it becomes available.
- [ ] Add Gitlab CI cross project testing for RAJAPerf on changes to RAJA develop branch
- [ ] Add Gitlab CI cross project testing for RAJAProxies on changes to RAJA develop
- [ ]  Add a "weekly" test pipeline that enables "exhaustive tests" and runs on the weekend. Note: resource allocation times will need to be bumped for these builds (not sure how to do that with our current specs setup, since we don't want to increase the times for regular builds)

**Comment by MrBurmark** [comment]
One idea is to make sure we test the default compiler modules as the defaults are often tied to what people are actually using.

**Comment by rhornung67** [comment]
Also address https://github.com/LLNL/RAJA/issues/1377

**Comment by rhornung67** [comment]
Also address https://github.com/LLNL/RAJA/issues/1330

**Comment by rhornung67** [comment]
will add new issue with updated information
